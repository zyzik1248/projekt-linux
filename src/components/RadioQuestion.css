.RadioQuestion {
  margin: 30px;
}

.RadioQuestion div{
  margin-top: 5px;
}

