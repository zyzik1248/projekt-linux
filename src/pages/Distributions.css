.Distributions {
  min-height: 100vh;
}

.Distributions-items {
  margin: auto;
  width: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.Distributions-items div:nth-child(2) {
  background: #f6f6f6;
}

.Distributions-items div:nth-child(3) {
  background: #f6f6f6;
}

.Distributions-items div:nth-child(6) {
  background: #f6f6f6;
}

.Distributions-item {
  height: 85vh;
  min-height: 550px;
  width: 50%;
}

@media only screen and (max-width: 900px) {
  .Distributions-item {
    width: 100%;
  }
}

.Distributions-item-content {
  margin: auto;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.Distributions-item img {
  height: 150px;
}

.Distributions-item h1 {
  font-family: sans-serif;
  line-height: 170%;
  font-size: 20px;
  padding-top: 10px;
  margin: 0;
  color: rgb(83, 83, 83);
}

.Distributions-item p {
  font-family: sans-serif;
  line-height: 170%;
  font-size: 16px;
  padding: 5px 50px;
  color: rgb(83, 83, 83);
}

@media only screen and (max-width: 450px) {
  .Distributions-item {
    min-height: 700px;
  }
  .Distributions-item p {
    padding: 5px 10px;
  }
}