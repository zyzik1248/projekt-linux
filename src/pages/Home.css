.Home{
  width: 100%;
  height: 100vh;
  background-image: url('/src/assets/Linux-Background.jpeg');
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: center;
}

.Home-info{
  width: 60%;
  padding: 70px;
  background-color: rgba(255, 255, 255, 0.8);
  text-align: center;
  color: #606060;
}

.Home-big-text{
  font-size: 32px;
}

.Home-small-text{
  font-size: 28px;
}