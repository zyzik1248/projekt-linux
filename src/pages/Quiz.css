.Quiz {
  min-height: 100vh;
  padding-top: 100px;
  background: linear-gradient( 180deg, rgb(11, 12, 14) 0%, rgb(25, 28, 32) 100%);
  width: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.Quiz h4 {
  font-size: 40px;
  color: #7510F7;
}

.Quiz button {
  background: none;
  padding: 10px 20px;
  color: white;
  border: 2px solid #7510F7;
  border-radius: 10px;
  margin: 20px;
  margin-bottom: 0;
  cursor: pointer;
}

.Quiz button:hover:enabled {
  background: #7510F7;

}

.Quiz button:disabled {
  border: 2px solid rgba(116, 16, 247, 0.527);
  color: rgba(255, 255, 255, 0.589);
}

.Quiz-good-text-info{
  background-color: rgba(0, 128, 0, 0.5);
  padding: 20px 40px;
  color: white;
}

.Quiz-wrong-text-info{
  background-color: rgba(255, 0, 0, 0.5);
  color: white;
  padding: 20px 40px;
}

.Quiz-input{
  margin: 20px;
}