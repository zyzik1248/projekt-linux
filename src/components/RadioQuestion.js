import React from 'react'

import './RadioQuestion.css'

class RadioAnswer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      answers: []
    };
  }

  componentDidMount() {
    let tab = this.props.answers.slice();
    const answers = [];

    for (let i = 0; i < this.props.answers.length; i++) {
      let id = Math.floor(Math.random() * tab.length);
      answers.push(
        tab[id] + ""
      )
      tab.splice(id, 1);
    }

    this.setState({ answers })
  }

  render() {
    return (
      <div className="RadioQuestion">
        {
          this.state.answers.map((el) =>
            <div key={el}>
              <input
                data-testid={`radio-question-${el}`}
                type="radio"
                id={el}
                name={`quiz_radio_${this.props.id}`}
                value={el}
                onChange={this.props.handleChange}
                disabled={this.props.disabled}
              />
              <label htmlFor={el}>{el}</label>
            </div>
          )}
      </div>
    )
  }
}

export default RadioAnswer;