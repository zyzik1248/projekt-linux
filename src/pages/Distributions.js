import React from 'react'

import './Distributions.css'
import distributions from './../assets/distributions.json'
import distributions2 from './../assets/distributions2.json'
import requireContext from 'require-context.macro';

class Home extends React.Component {
  constructor() {
    super();
    this.txtData = () => JSON.parse(JSON.stringify(distributions));
    this.txtData2 = () => JSON.parse(JSON.stringify(distributions2));
    const imgContext = requireContext('../assets/', false, /\.png$/);
    let img = {}
    this.imgs = imgContext.keys().reduce((icons, file) => {
      const cname = imgContext(file).default;
      const label = file.slice(2, -4);
      img[label] = cname;
      return img;
    }, {})
  }

  render() {
    const tabData = this.txtData().data; 

    const items = tabData.map((item) => ( //2.2 pierwszy sposób
      <div className='Distributions-item' key={item.title}>
        <div id={`distribution-${item.id}`} className="Distributions-item-content">
          <img alt={item.title} data-testid='logo_distr' src={this.imgs[item.img]} />
          <h1>{item.title}</h1>
          <p>{item.text}</p>
        </div>
      </div>
    ))

    const tabData2 = this.txtData2();

    const items2 = []; //2.2 drugi sposób
    for (let i = 0; i < tabData2.count; i++) {
      let value = tabData2.data[i]
      items2.push(
        <div className='Distributions-item' key={value.title}>
          <div className="Distributions-item-content">
            <img alt={value.title} data-testid='logo_distr' src={this.imgs['logo' + (i + 1)]} />
            <h1>{value.title}</h1>
            <p>{value.text}</p>
          </div>
        </div>
      )
    }

    return (
      <div id='distributions' className="Distributions">
        <div className='Distributions-items'>
          {items}
          {items2}
        </div>
      </div>
    )
  }
}

export default Home;