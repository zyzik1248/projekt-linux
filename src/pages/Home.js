import React from 'react'

import './Home.css'

class Home extends React.Component {
  render() {
    return (
      <div id='home' className='Home'>
        <div className='Home-info'>
          <p className='Home-big-text'>Wszystko o linuxie</p>
          <p className='Home-small-text'>
            Na tej stronie dowiesz siÄ™ ciekawych informacji na temat tego systemu
          </p>
        </div>
      </div>
    )
  }
}

export default Home;