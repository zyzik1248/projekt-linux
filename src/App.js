import './App.css';
import { BrowserRouter as Router } from "react-router-dom";

import Header from './components/Header'
import Navigation from './components/Navigation'
import Home from './pages/Home'
import History from './pages/History'
import Distributions from './pages/Distributions'
import Terminal from './pages/Terminal'
import Quiz from './pages/Quiz'

function App() {
  return (
    <Router className="App">
      <Header />
      <Navigation />
      <Home/>
      <History />
      <Terminal />
      <Distributions />
      <Quiz />
    </Router>
  );
}

export default App;
