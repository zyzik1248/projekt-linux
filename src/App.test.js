import React from 'react';
import { render, fireEvent, getByText } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { Router } from "react-router-dom";
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { render as ren, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import App from './App'
import Home from './pages/Home'
import Quiz from './pages/Quiz';
import RadioQuestion from './components/RadioQuestion'
import Distributions from './pages/Distributions'

test("renders components", async () => {
  const { getAllByText } = render(<App />)
  expect(getAllByText("Wszystko o linuxie")[0]).toBeInTheDocument()
})

test('click start quiz corectly', async () => {
  const { getByTestId } = render(<App />)
  fireEvent.click(getByTestId(/start quiz/i))
  expect(getByTestId(/b quiz/i)).toBeInTheDocument()
})

test("matches snapshot home page", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
})

test('should display timer', async () => {
  jest.useFakeTimers();
  const { getByTestId } = render(<Quiz />)
  fireEvent.click(getByTestId(/start quiz/i))
  jest.advanceTimersByTime(5000);
  expect(getByTestId(/timer/i)).toHaveTextContent("55")
  //sanity test
  expect(getByTestId(/timer/i)).not.toHaveTextContent("54")
})

test('should radio question work', () => {
  jest.useFakeTimers();
  const handleChange = jest.fn();
  const { getByTestId } = render(
    <RadioQuestion
      correct="ps aux"
      id={2}
      answers={["rm -fr", "cd", "chmod +rx", "ps aux"]}
      handleChange={handleChange}
      disabled={true} />
  )
  fireEvent.change(getByTestId(/radio-question-cd/i), { target: { value: "cd" } })
  jest.runAllTimers();
  expect(getByTestId(/radio-question-cd/i).value).toBe('cd')
  expect(getByTestId(/radio-question-cd/i).value).not.toBe('chmod +rx')
})

test('data from json file', async () => {
  const { getAllByTestId } = render(<Distributions />)
  expect(getAllByTestId('logo_distr').map((el) =>
    el.alt
  )).toContain('ubuntu')
})