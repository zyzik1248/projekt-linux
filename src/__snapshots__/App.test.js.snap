// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot home page 1`] = `
<div
  className="Home"
  id="home"
>
  <div
    className="Home-info"
  >
    <p
      className="Home-big-text"
    >
      Wszystko o linuxie
    </p>
    <p
      className="Home-small-text"
    >
      Na tej stronie dowiesz siÄ™ ciekawych informacji na temat tego systemu
    </p>
  </div>
</div>
`;
